### PR TITLE
resume download after fail

### LINKED ISSUE #2106: Resume ISO (and other) Downloads
I have a standalone CI builder (packer runner) that re-builds on commit. It often fails to download ISOs where wget can automatically resume.

I'd like to see either support for download retries / resuming or perhaps a configuration setting to use an external configurable download tool/command to better deal with these situations.

### PR DESCRIPTION
**Pull request by vtolstov**
close #2106

Signed-off-by: Vasiliy Tolstov <v.tolstov@selfip.ru>

### CODE REVIEW

**Comment by vtolstov** [comment]
Does this accepted by upstream ? I'm change Downloader....

File: common/download.go [hunk 1]
```diff
@@ -195,9 +197,9 @@ func (*HTTPDownloader) Cancel() {
 	// TODO(mitchellh): Implement
 }
 
-func (d *HTTPDownloader) Download(dst io.Writer, src *url.URL) error {
+func (d *HTTPDownloader) Download(dst *os.File, src *url.URL) error {
 	log.Printf("Starting download: %s", src.String())
-	req, err := http.NewRequest("GET", src.String(), nil)
+	req, err := http.NewRequest("HEAD", src.String(), nil)
```
Comment (mitchellh) [inline common/download.go]: Not all servers support HEAD. I think if this is the case, we should just fall back to using `GET` with no range.

**Comment by mitchellh** [comment]
LGTM! Moving downloader to work with files certainly makes this easier. I've made one comment, but otherwise LGTM!

**Comment by vtolstov** [comment]
May be this is not the best solution as i see =(. May be doing GET request with Range 0-1 and check that we have single byte more preferred solution, @mitchellh what do you think?

**Comment by mitchellh** [comment]
@vtolstov I like your solution, let's just fall back to disabling range downloads if the server doesn't support HEAD requests.

**Comment by vtolstov** [comment]
Ok, nice.

**Comment by mitchellh** [comment]
@vtolstov Let me know when its done, and I'll re-review

**Comment by vtolstov** [comment]
T think i'm ready

**Comment by mitchellh** [comment]
This LGTM. I'll merge shortly I want to test it out a bit locally.

**Comment by mitchellh** [comment]
I've also been working on unit tests for this, so if I can get those passing with this, that'd be icing.

**Comment by mitchellh** [comment]
I merged some tests, so I'm going to accept this, run the tests, and write one for this case as well.

**Pull request merged**